agent/tests/test_monitoring.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+import requests
+from .test_input.test_zpipeline_base import TestInputBase
+from .test_pipelines.test_zpipeline_base import TestPipelineBase
+
+
+class TestDirectory(TestInputBase, TestPipelineBase):
```

Comment:
I guess you need to add this test to the Makefile test section, it didn't run on the github action.
And maybe we can test it in the test directory pipeline? I'm not sure it's a good idea, but that way we won't need to create directory sources and pipelines two times.
And btw, you don't delete sources and pipelines and test input files will try to create them for a second time.
And btw, it will break other directory tests because these pipelines will send data, and those pipelines will send data for the second time and test output will fail

Reply:
Yes, firstly I thought to add into Directory tests, but then decided to create separate test (Directory sources\pipelines - because they do not require additional containers).
If in a new Makefile command we add also `bootstrap` command before execution this new test, it will not affect other tests, right?

Reply:
Maybe we can create a separate single pipeline for this test?